### PR TITLE
feat: added arima learner and fixed S3 missing argument warning

### PR DESCRIPTION
**Pull request by m-muecke**

### CODE REVIEW

File: tests/testthat/test_learner_arima.R [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+test_that("Basic Tests", {
+  task = tsk("airpassengers")
+  learner = lrn("forecast.arima")
+  learner$train(task, 1:20)
+  fitted_values = learner$fitted_values(row_ids = 5:10)
+  expect_data_table(fitted_values, nrows = 6, ncols = length(task$target_names), types = "numeric")
+  expect_prediction(learner$predict(task, 21:30))
+
+  resampling = ResamplingCustom$new()
+  resampling$instantiate(task, train_sets = list(1:100), test_sets = list(101:144))
```
Comment (sebffischer) [inline tests/testthat/test_learner_arima.R:19]: I wouldnt test the Resample Result but focus more on the Prediction object itself (if the learner API works then the ResampleResult should work as well)

File: tests/testthat/test_learner_arima.R [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  rr = resample(task, learner, resampling)
+  expect_resample_result(rr)
+
+  forecast = learner$forecast(task = task, h = 10, new_data = task$data(rows = 11:20, cols = "fdeaths"))
```
Comment (sebffischer) [inline tests/testthat/test_learner_arima.R:23]: set h to different value than default

File: tests/testthat/test_learner_arima.R [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  expect_prediction_forecast(forecast)
+})
+
+test_that("Exogenous Variables", {
+  task = TaskRegrForecast$new(id = "se", backend = tsbox:: ts_c(mdeaths, fdeaths), target = "mdeaths")
```
Comment (sebffischer) [inline tests/testthat/test_learner_arima.R:28]: where do mdeaths and fdeaths come from?

File: tests/testthat/test_learner_arima.R (hunk 1 above)
Comment (sebffischer) [reply tests/testthat/test_learner_arima.R:19]: e.g. check that the fitted values are correctly contained in the prediction.

Comment (sebffischer) [reply tests/testthat/test_learner_arima.R:19]: Also expect_learner should be run I guess

File: tests/testthat/test_learner_arima.R [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  expect_prediction(learner$predict(task))
+})
+
+test_that("five row, five col", {
+  data = data.frame(target = rnorm(5), col2 = rnorm(5))
+  task = TaskRegrForecast$new(id = "one row, two col", backend = ts(data), target = "target")
+  learner = lrn("forecast.arima")
+  learner$train(task)
+  expect_prediction(learner$predict(task))
+})
```
Comment (sebffischer) [inline tests/testthat/test_learner_arima.R:60]: What is also missing here is a parameter test to check that the learner implements the available parameters correctly: https://github.com/mlr-org/mlr3/blob/0189c24bbcf29fa59b25b89be718f1d3fb2268a5/tests/testthat/test_mlr_learners_classif_rpart.R#L9

**Review by sebffischer — COMMENTED** [review]
Added some comments. We can then iterate again :)

File: tests/testthat/test_learner_arima.R (hunk 3 above)
Comment (m-muecke) [reply tests/testthat/test_learner_arima.R:28]: come from datasets package

File: R/LearnerRegrForecastArima.R [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+      }
+      response = as.data.table(as.numeric(forecast$mean))
+      colnames(response) = task$target_names
+
+      se = as.data.table(as.numeric(
+        ci_to_se(width = forecast$upper[, 1] - forecast$lower[, 1], level = forecast$level[1])
+      ))
+      colnames(se) = task$target_names
+
+      truth = copy(response)
```
[the remark this answers is not in the document]
Comment (m-muecke) [reply R/LearnerRegrForecastArima.R:70]: copied from the other learners, the copy exists to overwrite the values with 0 hence have the same no. of rows, doesn't make much sense, but the `PredictionForecast` implication requires the truth argument, which is probably the reason

File: R/LearnerRegrForecastArima.R [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+      truth[, colnames(truth) := 0]
```
[the remark this answers is not in the document]
Comment (m-muecke) [reply R/LearnerRegrForecastArima.R:71]: copied from the other learners, the copy exists to overwrite the values with 0 hence have the same no. of rows, doesn't make much sense, but the `PredictionForecast` implication requires the truth argument, which is probably the reason

File: R/LearnerRegrForecastArima.R (hunk 5 above)
[the remark this answers is not in the document]
Comment (m-muecke) [reply R/LearnerRegrForecastArima.R:70]: copied from the other learners

**Pull request merged**